test/AutoRest.TestServer.Tests/TestMgmtParent.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using AutoRest.CSharp.AutoRest.Communication;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Output.Models.Types;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+
+namespace AutoRest.TestServer.Tests
+{
+    internal class TestMgmtParent
+    {
+        [Test]
+        public async Task TestParentComputer()
+        {
+            var basePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            basePath = Path.Combine(basePath.Substring(0, basePath.IndexOf("autorest.csharp")), "autorest.csharp", "samples", "Azure.ResourceManager.Sample", "Generated");
+            var configuration = StandaloneGeneratorRunner.LoadConfiguration(basePath, File.ReadAllText(Path.Combine(basePath, "Configuration.json")));
+            var codeModelTask = Task.Run(() => CodeModelSerialization.DeserializeCodeModel(File.ReadAllText(Path.Combine(basePath, "CodeModel.yaml"))));
+            var projectDirectory = Path.Combine(configuration.OutputFolder, Configuration.ProjectRelativeDirectory);
+            var project = await GeneratedCodeWorkspace.Create(projectDirectory, configuration.OutputFolder, configuration.SharedSourceFolders);
+            var sourceInputModel = new SourceInputModel(await project.GetCompilationAsync());
+            var model = await codeModelTask;
+            var context = new BuildContext<MgmtOutputLibrary>(model, configuration, sourceInputModel);
+            _ = context.Library; // gen lib
```

Comment:
@m-nash @allenjzhang this is what I was talking about. To propagate values you now need to do strange things that don't seem related.

Reply:
?? not sure I follow

Reply:
Same thing about mutable properties on OperationGroup being assigned as a side effect of accessing the `context.Library`. And dependent on the order of execution.

It also makes it very hard to test this new feature in isolation without loading the entire code model and building a library.

Reply:
this is a test case I don't think I follow

Reply:
Do you mean don't follow the test case of the problem I'm trying to describe?

Reply:
the problem being described

Reply:
2 problems:
1. this test has to run half of the generator to test a small piece of code because of the way things are coupled
2. The value of the `operations.Parent` property depends on the order of execution and side effects of other parts of the system.

For example:

``` C#
     var model = await codeModelTask;
            var context = new BuildContext<MgmtOutputLibrary>(model, configuration, sourceInputModel);
            _ = context.Library; // gen lib

model.OperationGroups.First().Parent
```

Would give you the correct value. Note that context and library were not used in any of the asserts so let's remove them and try the following code instead:

while 

``` C#
    var model = await codeModelTask;
    
	model.OperationGroups.First().Parent

```

Whoops, it doesn't work anymore.

Reply:
Ok, so basically you are re-asserting that this [item ](https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5724)we opened up needs to get fixed.  I agree.

Reply:
No, that one talks about models returned from the output library. THis is about the mutability of the CodeModel types that serve as an input to the OutputLibrary.

Reply:
ok, you are right this is the one that @allenjzhang talked about which again is something we are looking at separately.  This PR didn't cause it so I don't want to hold up this PR because of it

Reply:
Is there a tracking issue?